bin/proving-service/src/api/mod.rs

```diff
@@ -21,48 +24,133 @@ pub struct RpcListener {
 }
 
 impl RpcListener {
-    pub fn new(listener: TcpListener) -> Self {
-        let api_service = ApiServer::new(ProverRpcApi::default());
+    pub fn new(listener: TcpListener, is_tx_prover: bool, is_batch_prover: bool) -> Self {
+        let prover_rpc_api = ProverRpcApi::new(is_tx_prover, is_batch_prover);
+        let api_service = ApiServer::new(prover_rpc_api);
         Self { listener, api_service }
     }
 }
+struct Provers {
+    tx_prover: Option<LocalTransactionProver>,
+    batch_prover: Option<LocalBatchProver>,
+}
+
+impl Provers {
+    fn new(is_tx_prover: bool, is_batch_prover: bool) -> Self {
+        let tx_prover = if is_tx_prover {
+            Some(LocalTransactionProver::default())
+        } else {
+            None
+        };
+
+        let batch_prover = if is_batch_prover {
+            Some(LocalBatchProver::new(MIN_PROOF_SECURITY_LEVEL))
+        } else {
+            None
+        };
+
+        Self { tx_prover, batch_prover }
+    }
+}
```

Comment:
nit: This could be written like this for a more concise approach, though your approach might be more clear so feel free to disregard:

```rust
impl Provers {
    fn new(is_tx_prover: bool, is_batch_prover: bool) -> Self {
        Self {
            tx_prover: is_tx_prover.then(LocalTransactionProver::default),
            batch_prover: is_batch_prover.then(|| LocalBatchProver::new(MIN_PROOF_SECURITY_LEVEL)),
        }
    }
}

```

Reply:
Yeah, I thought about doing something like that but prefer to be more explicit. If you consider your approach clearer let me know and I can implement it.